File: src/main/java/com/mycompany/taller_v2/MaintenanceManagement/MaintenanceManager.java

package com.mycompany.taller_v2.MaintenanceManagement;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class MaintenanceManager {
    private final List<MaintenanceAdvance> maintenanceProgresses;
    private MaintenanceStatus maintenanceStatus;

    public MaintenanceManager() {
        this.maintenanceProgresses = new ArrayList<>();
        this.maintenanceStatus = MaintenanceStatus.PENDING; // Estado inicial
    }

    public void addMaintenanceProgress(Date date, String description, Image image) {
        this.maintenanceProgresses.add(new MaintenanceAdvance(date, description, image));
    }

    public List<MaintenanceAdvance> getMaintenanceProgresses() {
        return maintenanceProgresses;
    }

    // Métodos para manejar el estado de mantenimiento
    public MaintenanceStatus getMaintenanceStatus() {
        return maintenanceStatus;
    }

    public void setMaintenanceStatus(MaintenanceStatus newStatus) {
        this.maintenanceStatus = newStatus;
    }
}

File: taller-front/src/interfaces/MaintenanceManager.ts
import { MaintenanceAdvance } from "./MaintenanceAdvance";

export enum MaintenanceStatus {
  PENDING = "PENDING",
  IN_PROGRESS = "IN_PROGRESS",
  COMPLETED = "COMPLETED",
}

export interface MaintenanceManager {
  idMaintenanceManager: number;
  maintenanceProgresses: MaintenanceAdvance[];
  maintenanceStatus: MaintenanceStatus;
}
